### DISCUSSION TITLE
Drawbacks of 404 response in WebhookHandler (or any not 5xx response)

### DISCUSSION THREAD

**Discussion by varemel** [def-studio/telegraph#417] [General]
Hello,

_Short version:_

**Can you please clarify what is the reason to throw 404 error when unknown chats are not allowed in telegraph config?** As far as I understand it will increase your bot `pending_update_count`. So, telegram api will resend this updates to your webhook over and over again "reasonable amount" of times.
> In case of an unsuccessful request, we will give up after a reasonable amount of attempts. https://core.telegram.org/bots/api#setwebhook

---

_Details:_

Can't find any mention about accepted response codes. So, I tested Telegram Webhook API myself. 
It assumes your webhook responses with 2xx status code.
Any different response codes (3xx, 4xx, 5xx) result in `pending_update_count` increase.
I can't decide whether it's good to always return 204 response, except for unexpected server errors with 5xx response.
**What do you think about it?**

![005652](https://github.com/defstudio/telegraph/assets/84812777/c90956fd-7f39-4d6b-a53b-83ae6d2a0ddd)

`vendor/defstudio/telegraph/src/Handlers/WebhookHandler.php`
```
protected function setupChat(): void
    {
        $telegramChat = $this->message?->chat() ?? $this->callbackQuery?->message()?->chat();

        assert($telegramChat !== null);

        .....

        if (!$this->chat->exists) {
            if (!$this->allowUnknownChat()) {
===>             throw new NotFoundHttpException();
            }
            .....
    }
```

**Comment by fabio-ivona** [comment]
Hi! Thanks for pointing this out

Will have to investigate on this, it is an interesting issue

**Comment by tobya** [comment]
Can I add my voice to this.  I just spent 2 hours wondering why I was getting 404 only to realise that allow_messages_from_unknown_chats was set to false.
